songs_app_backend/songs/serializers.py

```diff
@@ -0,0 +1,108 @@
+from rest_framework import serializers
+from .models import Song, UserSongLike, UserSongFavorite, UserSongComment
+from taggit.serializers import TagListSerializerField, TaggitSerializer
+from .tasks import add_song_task
+from datetime import datetime, timedelta
+import pytz
+
+
+class SongSerializer(TaggitSerializer, serializers.ModelSerializer):
+    tags = TagListSerializerField()
+
+    class Meta:
+        model = Song
+        fields = ('id', 'title', 'singer', 'tags')
+
+
+class AddSongSerializer(TaggitSerializer, serializers.ModelSerializer):
+    tags = TagListSerializerField()
+
+    class Meta:
+        model = Song
+        fields = ('id', 'title', 'singer', 'tags', 'created_time')
+        extra_kwargs = {
+            'created_time': {'required': True}
+        }
+
+    def validate(self, data):
+        scheduled_time = data.get('created_time')
+        schedule_datetime = scheduled_time.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        # Get the current time in Pakistan Standard Time (PST)
+        pst_timezone = pytz.timezone("Asia/Karachi")
+        current_time = datetime.now(
+            pst_timezone).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        # Check if the scheduled time is in the future
+        if schedule_datetime <= current_time:
+            raise serializers.ValidationError(
+                "Scheduled time must be in the future.")
+
+        data["creation_time"] = self.calculate_creation_time_seconds(
+            schedule_datetime, current_time)
+        return data
```

Comment:
if validate is only being used for created_time there is no need for it as it'll be automatically handled